source/server/connection_handler_impl.cc

```diff
@@ -327,27 +333,28 @@ ConnectionHandlerImpl::ActiveConnection::ActiveConnection(ActiveTcpListener& lis
   connection_->addConnectionCallbacks(*this);
   listener_.stats_.downstream_cx_total_.inc();
   listener_.stats_.downstream_cx_active_.inc();
+  listener_.per_worker_stats_.downstream_cx_total_.inc();
+  listener_.per_worker_stats_.downstream_cx_active_.inc();
 }
 
 ConnectionHandlerImpl::ActiveConnection::~ActiveConnection() {
   listener_.stats_.downstream_cx_active_.dec();
   listener_.stats_.downstream_cx_destroy_.inc();
+  listener_.per_worker_stats_.downstream_cx_active_.dec();
   conn_length_->complete();
 }
 
-ListenerStats ConnectionHandlerImpl::generateStats(Stats::Scope& scope) {
-  return {ALL_LISTENER_STATS(POOL_COUNTER(scope), POOL_GAUGE(scope), POOL_HISTOGRAM(scope))};
-}
-
-ActiveUdpListener::ActiveUdpListener(Event::Dispatcher& dispatcher, Network::ListenerConfig& config)
-    : ActiveUdpListener(dispatcher.createUdpListener(config.socket(), *this), config) {}
+ActiveUdpListener::ActiveUdpListener(Network::ConnectionHandler& parent,
+                                     Event::Dispatcher& dispatcher, Network::ListenerConfig& config)
+    : ActiveUdpListener(parent, dispatcher.createUdpListener(config.socket(), *this), config) {}
 
-ActiveUdpListener::ActiveUdpListener(Network::ListenerPtr&& listener,
+ActiveUdpListener::ActiveUdpListener(Network::ConnectionHandler& parent,
+                                     Network::ListenerPtr&& listener,
                                      Network::ListenerConfig& config)
-    : ConnectionHandlerImpl::ActiveListenerImplBase(std::move(listener), config),
-      udp_listener_(dynamic_cast<Network::UdpListener*>(listener_.get())), read_filter_(nullptr) {
+    : ConnectionHandlerImpl::ActiveListenerImplBase(parent, std::move(listener), config),
+      udp_listener_(*dynamic_cast<Network::UdpListener*>(listener_.get())), read_filter_(nullptr) {
   // TODO(sumukhs): Try to avoid dynamic_cast by coming up with a better interface design
-  ASSERT(udp_listener_ != nullptr, "");
+  ASSERT(dynamic_cast<Network::UdpListener*>(listener_.get()) != nullptr, "");
```

Comment:
Because the member variable is now a reference. It's possible that taking the address of the reference and checking for nullptr would do the same thing but I wasn't sure so I kept the assert as it originally was.

Reply:
I'm actually wondering if this assert does anything now, since you're dereferencing in the initializer list above. If the dynamic cast fails, you'll have dereferenced null.

Reply:
Correct, but it should never happen, and we would still assert here.

Reply:
Thanks for explanation! This ASSERT can make sure udp_listener_ as a reference is valid from that point which makes sense to me.